---
title: 'Principal Component Analysis in R for RNA-sequencing data'
date: 2052-02-24
permalink: /posts/2022/01/corr-PCA-post/
tags:
  - statistics
  - principal component analysis
  - R
---

### Introduction

Principal component analysis is often one of the first steps involved in the analysis of high dimensional data. These types of data have...well...a lot of dimensions, possibly hundreds or maybe thousands. And this is part of what makes analyzing this data so daunting at first glance.

RNA sequencing data is a good example of such daunting data. The human genome has roughly 20,000 genes, and depending on the cell type you are looking at a good portion of these genes might be detected by RNAseq. Often this is returned as individual rows of a matrix, each of which could be expressed at varying levels in an individual sample (represented by normalized count values that are organized into columns representing a single sample). Looking at each one these 20k rows across all the samples in your data set just isn't feasible sometimes. So what do you do?

### Dimensionality Reduction

There are several reasons that one of the first things you might want to do with your RNA-sequencing data is to reduce the number of dimensions using something like principal component analysis.

You may want to:

* Look for technical outliers
* Quickly assess the within-group variance of experimental replicates (i.e do biologically similar samples group together?)
* Quantify how much variation in your data set can be explained by technical factors (such as sequencing depth or batches)

I'm not going to go into the theory behind the linear algebra that is running under the hood of the different dim reduction techniques here, but instead I'll leave you with an analogy that I really like.

#### Coke Can Analogy

Imagine you wanted to know

### Normalization

### PCA

### Interpreting Results

### Correlating meta-data variables with latent variables

### Corrections (batch)
