---
title: 'Correlating Variables with Principal Components in R for RNA-sequencing data'
date: 2052-02-24
permalink: /posts/2022/01/corr-PCA-post/
tags:
  - statistics
  - principal component analysis
  - R
---

### Introduction

Principal component analysis is often one of the first steps involved in the analysis of high dimensional data. These types of data have...well...a lot of dimensions, possibly hundreds or maybe thousands. And this is part of what makes analyzing this data so daunting at first glance.

RNA sequencing data is a good example. The human genome has roughly 20,000 genes, often returned as individual rows of a matrix in pre-processed RNAseq results, each of which could be expressed at varying levels in an individual sample (represented by normalized count values that are organized into columns representing a single sample). Looking at each one these 20k rows across all the samples in your data set just isn't feasible sometimes. So what do you do?

### Dimensionality Reduction

There are several reasons that one of the first things you might want to do with your RNA-sequencing data is to reduce the number of dimensions using something like principal component analysis. You may want to:

* Look for technical outliers
* Assess the consistency of experimental replicates
* Quantify how much variation in your dataset can be explained by technical factors such as sequencing depth or batches
