---
title: "Distinct transcriptomic and epigenomic modalities underpin human memory T cell subsets and their activation potential"
collection: publications
permalink: /publication/2023-02-02-human-memory-T-1.md
excerpt: 'Integrating RNA-seq and ATAC-seq data to define gene regulatory networks associated with memory T cell subsets'
date: 2023-03-03
venue: 'Communications Biology'
paperurl: 'https://www.nature.com/commsbio/'
citation: 'James R. Rose,Bagdeser Akdogan-Ozdilek, Andrew R. Rahmberg,Michael D. Powell,Sakeenah L. Hicks,Christopher D. Scharer, Jeremy M. Boss. Communications Biology (In review)'
---

Human memory T cells (MTC) are poised to rapidly respond to antigen re-exposure.  Here, we derived the transcriptional and epigenetic programs of resting and ex vivo activated, circulating CD4+ and CD8+ MTC subsets.  A progressive gradient of gene expression from na√Øve to TCM to TEM was observed, which was accompanied by corresponding changes in chromatin accessibility.  Transcriptional changes suggested adaptations of metabolism that were reflected in altered metabolic capacity.  Other differences involved regulatory modalities comprised of discrete accessible chromatin patterns, transcription factor binding motif enrichment, and evidence of epigenetic priming.  Basic-helix-loop-helix factor motifs for AHR and HIF1A distinguish subsets and predict transcription networks to sense environmental changes.  Following stimulation, primed accessible chromatin correlated with an augmentation of MTC gene expression as well as effector transcription factor gene expression.  These results identify coordinated epigenetic remodeling, metabolic, and transcriptional changes that enable MTC subsets to ultimately respond to antigen re-encounters more effciently.

